src/client/containers/ProductPage/ProductPage.Container.js

```diff
@@ -1,12 +1,79 @@
+/* eslint-disable no-console */
+/* eslint-disable radix */
 import React from 'react';
+import { useParams } from 'react-router-dom';
+import { ProductDetails } from '../../components/ProductDetails/ProductDetails.component';
+import Carousel from '../../components/Carousel/Carousel.component';
+import ButtonComponent from '../../components/Button/Button.component';
 import './ProductPage.Style.css';
 
 const ProductPageContainer = () => {
+  const { id } = useParams();
+  const [product, setProduct] = React.useState({});
+  const [similarProduct, setSimilarProduct] = React.useState([]);
+
+  React.useEffect(() => {
+    fetch(`api/products/${id}`)
+      .then((res) => res.json())
+      .then((data) => {
+        setProduct(data[0]);
+      })
+      .catch((e) => {
+        console.log(e);
+      });
+  }, [id]);
+
+  React.useEffect(() => {
+    fetch(`api/categories/${product.category_id}`)
+      .then((res) => res.json())
+      .then((category) => {
+        console.log(category);
+
+        fetch(`api/products?category=${category[0].name}`)
```

Comment:
Use the custom hook: `useFetchApi`.

Reply:
I have explained it above.. I am not sure how to use useFetchApi hook in this usecase

Reply:
see my comment above

Reply:
done with useFetchAp